k256/src/ecdsa.rs

```diff
@@ -309,6 +371,22 @@ mod tests {
             }
         }
 
+        #[test]
+        fn public_key_recovery_zkvm() {
+            for vector in RECOVERY_TEST_VECTORS {
+                let digest = Sha256::new_with_prefix(vector.msg);
+                let sig = Signature::try_from(vector.sig.as_slice()).unwrap();
+                let recid = vector.recid;
+                let pk = <ecdsa_core::VerifyingKey<Secp256k1> as PowdrVerifyKey>::powdr_recover_from_prehash(
+                    &digest.finalize(),
+                    &sig,
+                    recid,
+                )
+                .unwrap();
```

Comment:
```suggestion
                let pk = VerifyingKey::powdr_recover_from_prehash(&digest.finalize(), &sig, recid)
                    .unwrap();
```
This works! You'd also need to remove the import of `Secp256k1` above.